product-service/src/test/java/com/projectwork/product_service/controller/ProductControllerTest.java

```diff
@@ -0,0 +1,94 @@
+
+
+@WebMvcTest(ProductController.class)
+@ExtendWith(SpringExtension.class)
+public class ProductControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private ProductService productService;
+
+    @Test
+    void shouldCreateProductAndReturnProductSuccessfullyCreated() throws Exception {
+        ProductRequest productRequest = getProductRequest();
+        String productRequestString = objectMapper.writeValueAsString(productRequest);
+        MvcResult mvcResult = mockMvc.perform(MockMvcRequestBuilders.post("/api/product").contentType(MediaType.APPLICATION_JSON)
+                        .content(productRequestString))
+                .andExpect(status().isCreated())
+                .andReturn();
+        Assertions.assertEquals(HttpStatus.CREATED.value(), mvcResult.getResponse().getStatus());
+    }
+
+    private ProductRequest getProductRequest() {
+        return ProductRequest.builder()
+                .name("iPhone")
+                .description("iphone 13")
+                .price(BigDecimal.valueOf(100))
+                .build();
+    }
+    @Test
+    void shouldReturnAllListOfProducts() throws Exception {
+        Mockito.when(productService.getAllProducts()).thenReturn(getProductResponse());
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/product").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0].name").value("iPhone"))
+                .andExpect(jsonPath("$[0].description").value("iphone 13"))
+                .andExpect(jsonPath("$[0].price").value(100))
+                .andExpect(jsonPath("$[1].name").value("iPhone"))
+                .andExpect(jsonPath("$[1].description").value("iphone 12"))
+                .andExpect(jsonPath("$[1].price").value(95));
+    }
+    private List<ProductResponse> getProductResponse() {
+        ProductResponse product1 = ProductResponse.builder()
+                .name("iPhone")
+                .description("iphone 13")
+                .price(BigDecimal.valueOf(100))
+                .build();
+
+        ProductResponse product2 = ProductResponse.builder()
+                .name("iPhone")
+                .description("iphone 12")
+                .price(BigDecimal.valueOf(95))
+                .build();
+
+        return List.of(product1, product2);
+    }
+    @Test
+    void shouldReturnNoProductsFoundInProductService() throws Exception {
+        Mockito.when(productService.getAllProducts()).thenReturn(productResponseList());
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/product").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+    private List<ProductResponse> productResponseList() {
+        List<ProductResponse> productResponseList = new ArrayList<>();
+        return productResponseList;
+    }
```

Comment:
nagative scenarios related to validation, say name or description is missing in the request body